### PR TITLE
updates docs for supported masking strategies and associated configs

### LINKED ISSUE #155: Update masking strategies guide with latest supported options
As of today our [How-To: Configure Masking Strategies](https://ethyca.github.io/fidesops/guides/masking_strategies/) isn't accurate, as it says this:

> Supported by fidesops policies
> - NullMaskingStrategy: Masks the input value with a null value.
> - ... More strategies coming soon

I believe we support all masking strategies in the policies now too (thanks to #108 and #119), so we can update the docs to reflect that.

### PR DESCRIPTION
**Pull request by eastandwestwind**
# Purpose
Update docs to reflect current supported masking strategies

# Changes
- Adds to `masking_strategies.md`. Screenshot of page for ease of review:

screencapture-0-0-0-0-8000-fidesops-guides-masking-strategies-2022-01-25-12_34_57

# Checklist

- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md)
- [x] Good unit test/integration test coverage

# Ticket

Fixes #155

### CODE REVIEW

File: docs/fidesops/docs/guides/masking_strategies.md [hunk 1]
```diff
@@ -80,27 +78,11 @@ The email has been replaced with a random string of 20 characters, while still p
...
-  - `HashMaskingStrategy`: Masks the input value by returning a hashed version of the input value
-  - `RandomStringRewriteMaskingStrategy`: Masks the input value with a random string of a specified length
-  - `AesEncryptionMaskingStrategy`: Masks by encrypting the value using AES
-  - `HmacMaskingStrategy`: Masks the input value by using the HMAC algorithm along with a hashed version of the data and a secret key
-
 ## Configuration
 
-Only null value masking is currently supported by fidesops policies, but support for other strategies is coming.
-Currently, erasure requests will replace customer data with null values.
+Erasure requests will replace mask data with the chosen masking strategy.
```
Comment (seanpreston) [inline docs/fidesops/docs/guides/masking_strategies.md:83]: I think we meant to remove either `mask` or `replace` in this line?

File: docs/fidesops/docs/guides/masking_strategies.md [hunk 2]
```diff
@@ -123,6 +105,77 @@ Issue a PATCH request to `/policy/policy_key/rule`:
...
+
+`configuration`:
+
+- `rewrite_value`: `str` that will replace input values
+- `format_preservation` (optional): `Dict` with the following key/vals:
+    - `suffix`: `str` that specifies suffix to append to masked value
+
+### Hash
+
+Masks the input value by returning a hashed version of the input value. Is deterministic such that the same input value will mask to the same value within the same privacy request. This is not the case when the masking service is called as standalone service (outside of a privacy request).
```
Comment (seanpreston) [inline docs/fidesops/docs/guides/masking_strategies.md:132]: Consider:
```
Masks the data by hashing the input before returning it. The hash is deterministic such that the same input will return the same output within the context of the same privacy request. This is not the case when the masking service is called as a standalone service, outside the context of a privacy request.
```

File: docs/fidesops/docs/guides/masking_strategies.md [hunk 3]
```diff
@@ -123,6 +105,77 @@ Issue a PATCH request to `/policy/policy_key/rule`:
...
+
+`configuration`:
+
+- `length` (optional): `int` that specifies length of randomly generated string. Default = `30`
+- `format_preservation` (optional): `Dict` with the following key/vals:
+    - `suffix`: `str` that specifies suffix to append to masked value
+
+### AES Encrypt
+
+Masks by encrypting the value using AES. Is deterministic such that the same input value will mask to the same value within the same privacy request. This is not the case when the masking service is called as standalone service (outside of a privacy request).
```
Comment (seanpreston) [inline docs/fidesops/docs/guides/masking_strategies.md:156]: As above for deterministic explanation

File: docs/fidesops/docs/guides/masking_strategies.md [hunk 4]
```diff
@@ -123,6 +105,77 @@ Issue a PATCH request to `/policy/policy_key/rule`:
...
+
+`configuration`:
+
+- `mode` (optional): `str` that specifies AES encryption mode. Only supported option is `GCM`. Default = `GCM`
+- `format_preservation` (optional): `Dict` with the following key/vals:
+    - `suffix`: `str` that specifies suffix to append to masked value
+
+### HMAC
+
+Masks the input value by using the HMAC algorithm along with a hashed version of the data and a secret key. Is deterministic such that the same input value will mask to the same value within the same privacy request. This is not the case when the masking service is called as standalone service (outside of a privacy request).
```
Comment (seanpreston) [inline docs/fidesops/docs/guides/masking_strategies.md:168]: As above for deterministic explanation

**Review by seanpreston — CHANGES_REQUESTED** [review]
A few small language tweaks but a very useful addition nonetheless, thanks @eastandwestwind

**Comment by eastandwestwind** [comment]
@seanpreston thanks for the comments! I've made those updates

**Review by pattisdr — APPROVED** [review]
great updates @eastandwestwind 🏆

**Review by seanpreston — APPROVED** [review]

**Pull request merged**